eStore/Core/Networking/ProductsRepository.swift

```diff
@@ -0,0 +1,110 @@
+//
+//  ProductsRepository.swift
+//  eStore
+//
+//  Created by Vladislav Kondrashkov on 11/17/19.
+//  Copyright © 2019 Vladislav Kondrashkov. All rights reserved.
+//
+
+import Foundation
+import Moya
+import ObjectMapper
+import Result
+
+enum ProductsRepositoryError: Error {
+    case badRequest
```

Comment:
Why do you expose internals of repisoty implementation? Keep it simple - `invalidInput` or / and `failed`